metabrainz.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+"""
+Author : Salvador <salvador.estrella.ortiz@gmail.com>
+Date   : 2022-09-30
+Purpose: Query Metadata from MusicBrainz API
+"""
+
+import argparse
+import requests
+from pprint import pprint
+
+
+# --------------------------------------------------
+def get_args():
+    """Get command-line arguments"""
+
+    parser = argparse.ArgumentParser(
+        description="Retrieve metadata from MusicBrainz API",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-a",
+        "--artist",
+        help="An artist to search for",
+        metavar="str",
+        type=str,
+        nargs="+",
+        required=True,
+    )
+
+    return parser.parse_args()
+
+
+# --------------------------------------------------
+class MusicMetadata:
+    artist_endpoit = "http://musicbrainz.org/ws/2/artist/"
+    release_groups_endpoint = "http://musicbrainz.org/ws/2/release-group/"
+    releases_endpoint = "http://musicbrainz.org/ws/2/release/"
+
+    def __init__(self, artist: list[str]):
+        self.artist = artist
+
+    def pick_artist(self, artist_list):
+        for artist in artist_list:
+            if artist["score"] == 100:
+                return artist
+
+    def query_artist(self) -> dict:
+        endpoint = MusicMetadata.artist_endpoit
+        artist_query = "%20".join(self.artist)
+        query = f"{endpoint}?query={artist_query}&limit=3&fmt=json"
+        response = requests.get(query)
+        artist_list = response.json()["artists"]
+        my_artist = self.pick_artist(artist_list)
+        pprint(my_artist)
```

Comment:
**suggestion:** Use the `logging` library instead.

Reply:
**question**: What would be the purpose of using the `logging` library? 

I only use `pprint` for testing purposes. maybe it'll be a good idea to remove those pprint statements when comitting? The final client won't print the response.

Reply:
The idea of having logs in your application is that you can debug or get useful runtime information if needed. For example, if you know that your program is going to execute 3 different functions you could log a message whenever the function is executed. Then if something fails you can change the logger level to debug and see all the messages and double check that your program is in fact executing all three functions.

Example:
Imagine that I want main to execute function a, and function a executes function b, and b executes c. But the program is not actually executing everything. A way of debugging this is by adding print statements, other is using breakpoints, but what happens if is someone else that is executing your program? How would you be able to help them without having to jump into a call? You could ask them to set the logger level to debug, and that they send you all the logs your program is generating so that you can inspect it in hopes you find something weird.

In this example, you would note in the logs that function c is never executed, but function b is. So now you have a lead on where to find the bug.

```python
import logging

logger = logging.getLogger(__name__)

def func_a():
  logger.debug("Executing A")
  func_b()

def func_b():
  logger.debug("Executing B")
  return  # Whoops! I "forgot" to remove this return!
  func_c()

def func_c():
logger.debug("Executing C")

def main():
  logger.debug("Executing main")
  func_a()

main()
```